plugins/plugin-jenkins/codenvy-plugin-jenkins-webhooks/src/main/java/com/codenvy/plugin/jenkins/webhooks/JenkinsConnector.java

```diff
@@ -0,0 +1,256 @@
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.io.OutputStream;
+import java.io.Reader;
+import java.io.StringWriter;
+import java.net.HttpURLConnection;
+import java.net.URL;
+import java.util.Base64;
+import java.util.Map;
+import java.util.Optional;
+
+import static com.google.common.base.Strings.isNullOrEmpty;
+import static javax.ws.rs.HttpMethod.GET;
+import static javax.ws.rs.HttpMethod.POST;
+import static javax.ws.rs.core.MediaType.APPLICATION_XML;
+import static javax.xml.transform.TransformerFactory.newInstance;
+import static org.eclipse.che.commons.lang.IoUtil.readAndCloseQuietly;
+
+/**
+ * Client for Jenkins API.
+ * One {@link JenkinsConnector} is configured for one Jenkins job.
+ *
+ * @author Stephane Tournie
+ * @author Igor Vinokur
+ */
+public class JenkinsConnector {
+
+    private static final Logger LOG = LoggerFactory.getLogger(JenkinsConnector.class);
+
+    private static final String JENKINS_CONNECTOR_PREFIX_PATTERN    = "env.CODENVY_JENKINS_CONNECTOR_.+";
+    private static final String JENKINS_CONNECTOR_URL_SUFFIX        = "_URL";
+    private static final String JENKINS_CONNECTOR_JOB_NAME_SUFFIX   = "_JOB_NAME";
+    private static final String JENKINS_CONNECTOR_FACTORY_ID_SUFFIX = "_FACTORY_ID";
+    private static final String PROPERTY_NOT_FOUND_ERROR_MESSAGE    = "No connector properties were found for url %s and job %s";
+    private static final String FACTORY_LINK_DISPLAYED_MESSAGE      = "factory link %s already displayed in description of Jenkins job %s";
+
+    @VisibleForTesting
+    String url;
+
+    private final String                  jobName;
+    private final ConfigurationProperties configurationProperties;
+
+    @Inject
+    public JenkinsConnector(@Assisted("url") String url,
+                            @Assisted("jobName") String jobName,
+                            ConfigurationProperties configurationProperties) throws ServerException {
+        this.url = url;
+        this.jobName = jobName;
+        this.configurationProperties = configurationProperties;
+    }
+
+    /**
+     * Returns {@link JenkinsConnector} object with updated url from connector url property that contains credentials inside.
+     */
+    JenkinsConnector updateUrlWithCredentials() throws ServerException {
+        Optional<String> propertyPrefixOptional = getJenkinsConnectorPropertyPrefix();
+        if (propertyPrefixOptional.isPresent()) {
+            Map<String, String> properties = configurationProperties.getProperties(JENKINS_CONNECTOR_PREFIX_PATTERN);
+            this.url = properties.get(propertyPrefixOptional.get() + JENKINS_CONNECTOR_URL_SUFFIX);
+        } else {
+            throw new ServerException(String.format(PROPERTY_NOT_FOUND_ERROR_MESSAGE, url, jobName));
+        }
+        return this;
+    }
+
+    /**
+     * Returns related factory Id, configured for current connector by connector properties.
+     */
+    String getFactoryId() throws ServerException {
```

Comment:
This method is used in `JenkinsWebhooksManager` to get configured factoryId for given Jenkins url and job name. I think this method will be useless for other classes that may use the connect only for remote operations.